1-Computer-Vision/Exercises/1-3-2-Stochastic-Gradient-Descent/2022-08-29-Stochastic-Gradient-Descent.ipynb

```diff
@@ -46,7 +46,7 @@
   },
```

Comment:
<p>Looks like mean validation accuracy dropped slightly in your experiment run.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/jonathanloganmoran/ND0013-Self-Driving-Car-Engineer/pull/52/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='92'/>

Reply:
<p>That happens due to random initialisation of model parameters. Rerun it several times and the curves are totally different.</p>

Reply:
<p>Would setting a fixed global seed help address this? Assuming we can call <code>tf.random.set_seed</code> prior to initialising the model weight vector <code>W</code>, as shown below:</p><p> </p><h3><code># Setting a fixed global seed prior to initialising model weight vector</code></h3><h3><code>tf.random.set_seed(seed=42)</code></h3><p><code>W = tf.Variable(tf.random.normal(</code></p><p><code>        shape=(</code></p><p><code>            model_params['num_inputs'],&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;           model_params['num_outputs']</code></p><p><code>        ),&nbsp;</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;mean=0,&nbsp;</code></p><p><code>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;stddev=0.01</code></p><p><code>))</code></p>

Reply:
<p>Yes, setting a global seed is a choice. I am not familar with TensorFlow, but with PyTorch, you need to fix the seed of numpy as well, and enable the deterministic mode as well. </p><p> </p><p>Also, fixing the seed is usually used when comparing the performance of multiple models. Usually seed search should be performed as well to find the best model. Here I think it is fine to train the model without fixed random seed.</p>